### PR TITLE
Add FXIOS-5723 [v112] Detect NSLog, print and os_log in PRs with Danger

### PR DESCRIPTION
**Pull request by lmarceau**
## [FXIOS-5723](https://mozilla-hub.atlassian.net/browse/FXIOS-5723) https://github.com/mozilla-mobile/firefox-ios/issues/13179
Detect `NSLog()`, `print()` and `os_log()` in PRs with Danger. This will write a comment on the PR if there's a `print("example")` for example in a PR. This won't fail the PR. Removing code is not part of this check, so this only applies to new lines of code.

Let me know if you think of any ways of improving this message!

### Example of output
Generated from test PR [here](https://github.com/mozilla-mobile/firefox-ios/pull/13245)
```
Print() function seems to be used in file Client/Frontend/TabContentsScripts/NightModeHelper.swift at line 50. Please remove this usage from production code or use BrowserKit Logger.

NSLog() function seems to be used in file Client/Frontend/TabContentsScripts/NightModeHelper.swift at line 51. Please remove this usage from production code or use BrowserKit Logger.

os_log() function seems to be used in file Client/Frontend/TabContentsScripts/NightModeHelper.swift at line 52. Please remove this usage from production code or use BrowserKit Logger.
```

### CODE REVIEW

**Review by isabelrios — APPROVED** [review]
LGTM
My only concern is that the message in the PR's comment is ignored. We might need to document the errors / messages that danger posts and what to do...

**Review by nishant2718 — APPROVED** [review]

**Review by thatswinnie — APPROVED** [review]

**Review by yoanarios — APPROVED** [review]

**Comment by lmarceau** [comment]
> LGTM My only concern is that the message in the PR's comment is ignored. We might need to document the errors / messages that danger posts and what to do...

Yeah could be, but then it's on the reviewer's end as well to don't approve the PR if some things aren't respected. It's a first step to see how it goes, and then we can adjust if needed as you said!

**Comment by mte-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Edited 1 files</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Created 0 files</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 6719b96bfad7a50eabe45beacbe7a20452a95b98
</p>

**Review by OrlaM — APPROVED** [review]

**Comment by lmarceau** [comment]
Fixed swiftlint, not going to rerun BR which was [green](https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/9145e87b-b818-40c7-9625-325b48bcf7fd)

**Pull request merged**
